image/crypto/src/rustcrypto.rs

```diff
@@ -118,6 +118,24 @@ impl ImageGeneratorCrypto for RustCrypto {
 
         Ok(to_hw_format(&priv_key))
     }
+
+    fn sha512_digest(&self, data: &[u8]) -> anyhow::Result<ImageDigest512> {
+        let mut engine = Sha512::new();
+        engine.update(data);
+        Ok(to_hw_format(&engine.finalize()))
+    }
+
+    fn mldsa_pub_key_from_file(path: &Path) -> anyhow::Result<ImageMldsaPubKey> {
+        let key_bytes = std::fs::read(path)
+            .with_context(|| format!("Failed to read public key file {}", path.display()))?;
+        Ok(ImageMldsaPubKey(to_hw_format(&key_bytes)))
+    }
+
+    fn mldsa_priv_key_from_file(path: &Path) -> anyhow::Result<ImageMldsaPrivKey> {
+        let key_bytes = std::fs::read(path)
+            .with_context(|| format!("Failed to read private key file {}", path.display()))?;
+        Ok(ImageMldsaPrivKey(to_hw_format(&key_bytes)))
+    }
```

Comment:
nit: Since this doesn't actually depend on a crypto library I don't think we need to have separate implementations. Should this just be implemented on the trait directly instead of in each implementation?

Reply:
Good call.